docs/guides/grpc.md

```diff
@@ -0,0 +1,275 @@
+
+import "google/protobuf/empty.proto";
+
+package news;
+
+// Define message type for News with all its fields
+message News {
+    int32 id = 1;
+    string title = 2;
+    string body = 3;
+    string postImage = 4;
+}
+
+// Message with just id of single news
+message NewsId {
+    int32 id = 1;
+}
+
+// List of ids of news to get multiple response
+message MultipleNewsId {
+    repeated NewsId ids = 1;
+}
+
+// List of all news
+message NewsList {
+   repeated News news = 1;
+}
+
+// NewsService defines read and write operations for news items
+service NewsService {
+    // GetAllNews retrieves all news items without any arguments
+    rpc GetAllNews (google.protobuf.Empty) returns (NewsList) {}
+
+    // GetNews fetches a single news item by its ID
+    rpc GetNews (NewsId) returns (News) {}
+
+    // GetMultipleNews retrieves multiple news items based on their IDs
+    rpc GetMultipleNews (MultipleNewsId) returns (NewsList) {}
+}
+```
+
+### Implement gRPC service
+
+Now having the protobuf file you can write a server that implements `NewsService` at any language you want that supports gRPC.
+
+Tailcall organization has sample node.js service inside [this repo](https://github.com/tailcallhq/node-grpc) that you can pull to your local machine
+
+To spin up the sample service just run inside repo
+
+```sh
+npm i
+node server
+```
+
+and wait for logs about service is running.
+
+## Tailcall config
+
+Now when we have a running gRPC service we're going to write Tailcall's config to make the integration.
+
+To do this we need to specify GraphQL types corresponding to gRPC types we have defined in protobuf file. Let's create new file `grpc.graphql` file with following content:
+
+```graphql
+# The GraphQL representation for News message type
+type News {
+  id: Int
+  title: String
+  body: String
+  postImage: String
+}
+
+# Input type that is used to fetch news data by its id
+input NewsInput {
+  id: Int
+}
+
+# Resolves multiple news entries
+type NewsData {
+  news: [News]!
+}
+```
+
+Now when we have corresponding types in schema we want to define GraphQL Query that specifies the operation we can execute onto news. We can extend our config with next Query:
+
+```graphql
+type Query {
+  # Get all news i.e. NewsService.GetAllNews
+  news: NewsData!
+  # Get single news by id i.e. NewsService.GetNews
+  newsById(news: NewsInput!): News!
+}
+```
+
+Also let's specify options of Tailcall's ingress and egress in the beginning of the config using [`@server`](../operators/server.md) and [`@upstream`](../operators/upstream.md) operators.
+
+```graphql
+schema @server(port: 8000, graphiql: true) @upstream(baseURL: "http://localhost:50051", httpCache: true) {
+  query: Query
+}
+```
+
+To specify protobuf file to read types from, use `@link` operator with type `Protobuf` on schema. `id` is important part of the definition that will be used by `@grpc` operator later
+
+```graphql
+schema @link(id: "news", src: "./news.proto", type: Protobuf)
```

Comment:
```suggestion
schema @link(id: "news-id", src: "./news.proto", type: Protobuf)
```

we could change it to differentiate between the link id and the package name when setting the `@grpc` directive